lib/client-assets.php

```diff
@@ -185,10 +185,29 @@ function gutenberg_register_scripts_and_styles() {
 	gutenberg_register_packages_scripts();
 
 	// Inline scripts.
+	global $wp_scripts;
+	$wp_scripts->registered['wp-api-fetch']->deps[] = 'wp-hooks';
```

Comment:
This is a bit awkward in that the inline script has a dependency which the handle upon which it's attached does not. There's a pretty strong guarantee from the preceding lines that this would be registered, but it might be a good idea to have some protections here all the same, either with `isset` or by [`wp_script_is`](https://developer.wordpress.org/reference/functions/wp_script_is/).

Reply:
Likely not an issue, but I wonder if removing the dependency on `wp-hooks` from `api-fetch` runs the risk of breaking plugins that have created a dependency on `api-fetch` on the assumption it will also queue up `wp-hooks`.  Once could argue that's _doing it wrong_ anyways but I mentioned anyways just for consideration.

Reply:
We can't realistically hold ourselves hostage to never changing the dependencies of a core script because people aren't declaring their own dependencies correctly 🤷‍♂️

Reply:
> There's a pretty strong guarantee from the preceding lines that this would be registered, but it might be a good idea to have some protections here all the same, either with `isset` or by [`wp_script_is`](https://developer.wordpress.org/reference/functions/wp_script_is/).

I decided to be cautious and add a guarded condition in the rebased 65675b4f4108efbb450cabaa42d7408423bd96cb.